### PR TITLE
DAC6-461 - New navigator implementation.

### PR DESCRIPTION
**Pull request by iservport**

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -317,5 +317,5 @@ POST       /associated-enterprises/affected-by             controllers.enterpris
 GET        /associated-enterprises/change-affected-by      controllers.enterprises.IsAssociatedEnterpriseAffectedController.onPageLoad(mode: Mode = CheckMode)
 POST       /associated-enterprises/change-affected-by      controllers.enterprises.IsAssociatedEnterpriseAffectedController.onSubmit(mode: Mode = CheckMode)
 
-#GET        /associated-enterprises/check-answers           controllers.AssociatedEnterpriseCheckYourAnswersController.onPageLoad()
+GET        /associated-enterprises/check-answers           controllers.AssociatedEnterpriseCheckYourAnswersController.onPageLoad()
```
Comment (helensoniega) [inline conf/app.routes:320]: This can be removed. This is a duplicate.

File: app/controllers/individual/IndividualSelectAddressController.scala [hunk 2]
```diff
@@ -122,13 +123,7 @@ class IndividualSelectAddressController @Inject()(
...
               updatedAnswersWithAddress <- Future.fromTry(updatedAnswers.set(SelectedAddressLookupPage, addressToStore))
               _ <- sessionRepository.set(updatedAnswersWithAddress)
-            } yield {
-              if (redirectUsers) {
-                Redirect(routes.IndividualCheckYourAnswersController.onPageLoad())
-              } else {
-                Redirect(navigator.nextPage(IndividualSelectAddressPage, mode, updatedAnswersWithAddress))
-              }
-            }
+            } yield Redirect(redirect(mode, Some(value), 0, redirectUsers))
```
Comment (helensoniega) [inline app/controllers/individual/IndividualSelectAddressController.scala:126]: Index has a default of 0 too

File: app/controllers/organisation/EmailAddressQuestionForOrganisationController.scala [hunk 3]
```diff
@@ -90,13 +92,7 @@ class EmailAddressQuestionForOrganisationController @Inject()(
...
             updatedAnswers <- Future.fromTry(request.userAnswers.set(EmailAddressQuestionForOrganisationPage, value))
             _ <- sessionRepository.set(updatedAnswers)
-          } yield {
-            if (determineRoute) {
-              Redirect(routes.CheckYourAnswersOrganisationController.onPageLoad())
-            } else {
-              Redirect(navigator.nextPage(EmailAddressQuestionForOrganisationPage, mode, updatedAnswers))
-            }
-          }
+          } yield Redirect(redirect(mode, Some(value), 0, determineRoute))
```
Comment (helensoniega) [inline app/controllers/organisation/EmailAddressQuestionForOrganisationController.scala:95]: Same here

File: app/controllers/organisation/IsOrganisationAddressKnownController.scala [hunk 4]
```diff
@@ -89,13 +91,7 @@ class IsOrganisationAddressKnownController @Inject()(
...
             updatedAnswers <- Future.fromTry(request.userAnswers.set(IsOrganisationAddressKnownPage, value))
             _              <- sessionRepository.set(updatedAnswers)
-          } yield {
-            if (determineRoute) {
-              Redirect(routes.CheckYourAnswersOrganisationController.onPageLoad())
-            } else {
-              Redirect(navigator.nextPage(IsOrganisationAddressKnownPage, mode, updatedAnswers))
-            }
-          }
+          } yield Redirect(redirect(mode, Some(value), 0, determineRoute))
```
Comment (helensoniega) [inline app/controllers/organisation/IsOrganisationAddressKnownController.scala:94]: Same here

File: app/controllers/organisation/OrganisationAddressController.scala [hunk 5]
```diff
@@ -99,13 +101,7 @@ class OrganisationAddressController @Inject()(override val messagesApi: Messages
...
             updatedAnswers <- Future.fromTry(request.userAnswers.set(OrganisationAddressPage, value))
             _              <- sessionRepository.set(updatedAnswers)
-          } yield {
-            if (redirectUsers) {
-              Redirect(routes.CheckYourAnswersOrganisationController.onPageLoad())
-            } else {
-              Redirect(navigator.nextPage(OrganisationAddressPage, mode, updatedAnswers))
-            }
-          }
+          } yield Redirect(redirect(mode, Some(value), 0, redirectUsers))
```
Comment (helensoniega) [inline app/controllers/organisation/OrganisationAddressController.scala:104]: Same here

File: app/controllers/organisation/OrganisationNameController.scala [hunk 6]
```diff
@@ -84,13 +86,7 @@ class OrganisationNameController @Inject()(
...
             updatedAnswers <- Future.fromTry(userAnswers.set(OrganisationNamePage, value))
             _ <- sessionRepository.set(updatedAnswers)
-          } yield {
-            if (redirectUsers) {
-              Redirect(routes.CheckYourAnswersOrganisationController.onPageLoad())
-            } else {
-              Redirect(navigator.nextPage(OrganisationNamePage, mode, updatedAnswers))
-            }
-          }
+          } yield Redirect(redirect(mode, Some(value), 0, redirectUsers))
```
Comment (helensoniega) [inline app/controllers/organisation/OrganisationNameController.scala:89]: Same here

File: app/controllers/organisation/OrganisationSelectAddressController.scala [hunk 7]
```diff
@@ -129,13 +130,7 @@ class OrganisationSelectAddressController @Inject()(
...
               updatedAnswersWithAddress <- Future.fromTry(updatedAnswers.set(SelectedAddressLookupPage, addressToStore))
               _ <- sessionRepository.set(updatedAnswersWithAddress)
-            } yield {
-              if (redirectUsers) {
-                Redirect(routes.CheckYourAnswersOrganisationController.onPageLoad())
-              } else {
-                Redirect(navigator.nextPage(SelectAddressPage, mode, updatedAnswersWithAddress))
-              }
-            }
+            } yield Redirect(redirect(mode, Some(value), 0, redirectUsers))
```
Comment (helensoniega) [inline app/controllers/organisation/OrganisationSelectAddressController.scala:133]: Same here

**Comment by iservport** [comment]
Implemented in other tickets

**Pull request closed without merging**